### DISCUSSION TITLE
issues with Google cloud run and K8?

### DISCUSSION THREAD

**Discussion by dweng0** [runtime-env/import-meta-env#1023] [Q&A] [answered]
Hi all, has anyone else had trouble with getting `import.meta.env...` variables working in GCR? We have multiple services but one UI uses VITE, we have configured a server endpoint with an env variable of:

`const SERVER_ENDPOINT = import.meta.env.VITE_SERVER_ENDPOINT; `

**This works locally as a docker image** But, the endpoint is undefined on GCR.
It also works on other services that use older `dotenv` methods to retrieve environment variables.

My next process will be to use a different bundler/UI tool, (essentially one that doesn't use import.meta.env) but wanted to see if anyone else has come across this issue?

**Comment by TamarTal23** [comment]
did you solve this problem?

**Comment by soc221b** [comment]
Hello,

I apologize for the delay in responding to this issue.

To troubleshoot the issue you're facing with import.meta.env variables not working in GCR, it's essential to confirm whether the special expression is being replaced as expected. You can check this by comparing your code before and after the replacement:

```diff
- globalThis.import_meta_env = JSON.parse('"import_meta_env_placeholder"')
+ globalThis.import_meta_env = JSON.parse('{"NAME":"production"}');
```

If the replacement is not occurring as it should, you can execute the following command before serving your application to ensure the proper substitution:

```sh
npx import-meta-env -x .env.example
```

You can find a complete example in these links:

Dockerfile:
https://github.com/import-meta-env/import-meta-env/blob/479fe80959511ba88c54538f93d98ff59e94851a/packages/examples/docker-starter-example/Dockerfile#L29

Start Script:
https://github.com/import-meta-env/import-meta-env/blob/479fe80959511ba88c54538f93d98ff59e94851a/packages/examples/docker-starter-example/start.sh#L5-L8

I hope this helps you resolve the issue you're experiencing. If you have any further questions or require additional assistance, please don't hesitate to reach out.

**Reply by TamarTal23** [reply]
Hi, I've done as you've written but ultimately getting the following using react and vite:   The following variables were defined in .env.example.public file but are not defined in the environment

**Reply by soc221b** [reply]
Hello, @TamarTal23

This can happen when the environment variable key is misspelled.

If you are still experiencing this issue, we would like you to provide some code snippets for further investigation.

**Please also ensure that the code snippet does NOT contain sensitive information**

**Reply by TamarTal23** [reply]
The keys are spelled correctly as far as I can see.
Here is my client folder structure:
![image](https://github.com/import-meta-env/import-meta-env/assets/133848221/6fa7c0b9-66b5-4101-ab3f-ee7891ca83e9)

This is my start script in the dockerfile entrypoint:
![image](https://github.com/import-meta-env/import-meta-env/assets/133848221/31fc36d5-0794-4961-8090-6443f5f3e9ab)

In my .env.example.public file I wrote: PORT=
and in the index.html: 
![image](https://github.com/import-meta-env/import-meta-env/assets/133848221/4d74597f-a39b-41fb-bb95-84d347ca93bb)

I'm using Kubernetes and I add the file with env values in the pipelines, and the result in the machine is the following error:
![image](https://github.com/import-meta-env/import-meta-env/assets/133848221/383944f4-c04e-4d95-8e1e-441db2cbbbf0)

If you need any additional snippets let me know, thank you in advance! (:

**Reply by soc221b** [reply]
Thanks for your quick reply.

I found some information in the [official documentation](https://cloud.google.com/run/docs/configuring/services/environment-variables#reserved) and it says `PORT` is a reserved name.

Can you try another environment variable name? Thank you.

**Reply by TamarTal23** [reply]
Hi, tried with a different name and it didn't work 
![image](https://github.com/import-meta-env/import-meta-env/assets/133848221/453ff258-00ac-449d-bdd6-24a2a0cca7e9)

**Reply by soc221b** [reply]
Hi, could you add a log before running the conversion script? We can then differentiate whether the environment is passed to docker:

```diff
#!/bin/bash

+ echo ===
+ echo $TEST_ENV
+ echo ===

./import-meta-env-alpine -x .env.example -p dist/index.html || exit 1
```

Thank you 😢

**Reply by TamarTal23** [reply]
It's blank
![image](https://github.com/import-meta-env/import-meta-env/assets/133848221/c5968525-43af-40e1-9df6-c13b8d5d39de)

**Reply by soc221b** [reply, marked as answer]
It seems that the environment variables are not passed into the container.

I found this [article](https://medium.com/google-cloud/kubernetes-configmaps-and-secrets-68d061f7ab5b) for you.

hope it helps you.

**Reply by TamarTal23** [reply]
Thank you that really helped
